test_host.h

```diff
@@ -82,8 +83,10 @@ class TestHost {
  private:
   void SetupControl0() const;
   void SetupTextureStages() const;
-  static void SaveBackbuffer(const char *output_directory, const char *name);
-  void SaveZBuffer(const char *output_directory, const char *name) const;
+  static void EnsureFolderExists(const std::string &folder_path);
+  static std::string PrepareSaveFilePNG(std::string output_directory, const std::string &filename);
+  static void SaveBackBuffer(const std::string &output_directory, const std::string &name);
+  void SaveZBuffer(const std::string &output_directory, const std::string &name) const;
```

Comment:
Nit (and not something you introduced) - can this be static?

Reply:
uses non static member vars, left as is